### ISSUE TITLE
Jumps in classification scores for tracked queries during inference

### ISSUE THREAD

**Issue by kirilllzaitsev** [MCG-NJU/MeMOTR#31]
Hi,
I have trained a DAB-Memotr model on 3-frame sequences that achieves good results on train/val sets of a custom single-class dataset:

![Image](https://github.com/user-attachments/assets/9b5e553f-8ba9-4fe4-972f-73ca2ccf8bd6)

However, during inference, having tracked an object correctly during the past `t-K..t-1` timesteps, a tracking query may unexpectedly jump to a low classification score on frame `t` and return to a high score on `t+1`. For example, classification scores of 5 det + 1 track queries in three adjacent steps (note the frame `t` in the middle):

```
t-1: [0.0022, 0.0015, 0.0016, 0.0044, 0.0040, 0.9369]
t: [0.0039, 0.0066, 0.0324, 0.0087, 0.0215, 0.0105]
t+1: [0.0060, 0.0374, 0.0053, 0.0044, 0.0070, 0.9047]
```

The object is always present in this case. The problematic frame is more challenging than the adjacent ones due to motion blur, but the model was trained with motion blur augmentation: `A.MotionBlur(p=0.4, blur_limit=(3, 21))`. I'm wondering what could cause such jumps if the data is in the distribution of the train dataset (even if it weren't, I would expect the temporal continuity to play a role and not allow such jumps)

@HELLORPG , could you help with navigating the possible steps for troubleshooting?

I would like to ask about two other things. 

1. Could you elaborate on the role that temporal continuity played in your experiments from the paper qualitatively? For the aforementioned model for an in-hand object manipulation use case, I obtain the following picture with frame-to-frame motion of a single object on the test set:

![Image](https://github.com/user-attachments/assets/433c6a5e-0569-45da-bccb-17b31f7f1572)

Predicted motion has substantially more jitter than the groundtruth, while I would expect to observe some smoothing (smaller changes in velocities across frames) from using the memory.

2. The detection branch appears to struggle with partial occlusion, oftentimes detecting only the visible object part (although the dataset contains plenty of partially occluded samples). I'm wondering if resolving this is only a matter of scaling up the data to include more such cases

**Comment by HELLORPG** [comment]
I'm very sorry, I always thought I had already replied to this issue, so I missed it :cry:.

Regarding the confidence fluctuation issue you mentioned, I did not observe any significant occurrence in my experiments. It could be an issue with the size of the training data, or it might be that in my scenario, the motion blur problem is not as severe as in yours. I'm not sure if `MotionBlur` can perfectly simulate this data distribution, but intuitively, if the model has encountered enough appropriate cases during training, it should be able to provide reasonable confidence predictions. Perhaps you could provide more details.

About the other questions:
> Predicted motion has substantially more jitter than the groundtruth, while I would expect to observe some smoothing (smaller changes in velocities across frames) from using the memory.

In our model, we do not predict the motion of objects. So, are you using the track query to directly predict the objects' motion by yourself? If that's the case, I suspect that learnable motion prediction often requires a more complex design to achieve the same effectiveness as the Kalman filter. This is also why learnable motion prediction has not yet gained a dominant position in the MOT field.

> The detection branch appears to struggle with partial occlusion, oftentimes detecting only the visible object part (although the dataset contains plenty of partially occluded samples). I'm wondering if resolving this is only a matter of scaling up the data to include more such cases

Detection performance is indeed one of the drawbacks of DETR-based methods. Additionally, the image augmentation used during DETR training is not as effective as that of traditional detectors (like YOLOX). Perhaps you could try stronger data augmentation methods, which might be more efficient than simply expanding the dataset.

**Comment by kirilllzaitsev** [comment]
Hi. Thanks for your reply. The motion blur problem is depicted below:

![Image](https://github.com/user-attachments/assets/b2c5b72a-bd64-43b7-b6fa-a689293e24f8)

In this case, the confidence of a single tracking query for each frame is: `0.94, 0.01, 0.01, 0.9`. This problem is mitigated by scaling up the number of motion blur cases, but I am a bit surprised that the change is that dramatic given the long-memory lambda of `0.01`.

I predict absolute poses of objects and referred to motion as the delta in absolute pose between subsequent timesteps. Regarding the Kalman filter, this is an interesting comment with which I tend to agree based on my experiments so far.

Detection is indeed quite poor in occluded scenarios. At the same time, it boils down to the question of how temporal modeling can be useful here. Intuitively, I would expect a track query, initialized at a frame with a fully visible object, to capture its appearance/shape reliably, allowing for better occlusion performance at subsequent frames. However, I observed that detection results of the tracking queries are similar to those of single-frame detectors (the bbox is drawn mostly around the visible object part).

**Comment by HELLORPG** [comment]
> This problem is mitigated by scaling up the number of motion blur cases

This is expected. After all, our method is a data-driven model, and the distribution of data directly affects the final performance. 

> I am a bit surprised that the change is that dramatic given the long-memory lambda of `0.01`.

We inject the long-term memory into the track query, but do not directly add it to the output embedding. Therefore, in terms of approach, we have only intervened in the initialization of the query and cannot guarantee that it is fully transmitted to the final output embedding. This is also a data-driven process. 

> However, I observed that detection results of the tracking queries are similar to those of single-frame detectors (the bbox is drawn mostly around the visible object part).
 
I believe this is also a data-driven result. If you provide a complete bounding box annotation for the occluded object, I believe the predicted result should reflect that as well. As we shown below, on DanceTrack, our model can predict the complete bounding box for occluded dancers.

[![Image](https://github.com/user-attachments/assets/37de6b33-0f58-4c8e-aa11-a0d88ceeddc9)](https://github.com/MCG-NJU/MeMOTR/blob/main/assets/dancetrack_demo.gif)

**Comment by kirilllzaitsev** [comment]
@HELLORPG , thanks a lot for your comments, they are very helpful
